beets/importer.py

```diff
@@ -675,28 +675,36 @@ def align_album_level_fields(self):
         for item in self.items:
             item.update(changes)
 
-    def manipulate_files(self, move=False, copy=False, write=False,
-                         link=False, hardlink=False, session=None):
+    def manipulate_files(self, operation=None, write=False, session=None):
+        """ Copy, move, link or hardlink (depending on `operation`) the files
+        as well as write metadata.
+
+        `operation` should be an instance of `util.MoveOperation`.
+
+        If `write` is `True` metadata is written to the files.
+        """
+
         items = self.imported_items()
         # Save the original paths of all items for deletion and pruning
         # in the next step (finalization).
         self.old_paths = [item.path for item in items]
         for item in items:
-            if move or copy or link or hardlink:
+            if operation is not None:
                 # In copy and link modes, treat re-imports specially:
                 # move in-library files. (Out-of-library files are
                 # copied/moved as usual).
                 old_path = item.path
-                if (copy or link or hardlink) and self.replaced_items[item] \
-                   and session.lib.directory in util.ancestry(old_path):
+                if (operation != MoveOperation.MOVE
+                        and self.replaced_items[item]
+                        and session.lib.directory in util.ancestry(old_path)):
```

Comment:
flake8 has a (pretty annoying) suggestion here: it wants the `and`s to go on the line preceding, rather than the line following. Like this:

```py
                if (operation != MoveOperation.MOVE and
                        self.replaced_items[item] and
                        session.lib.directory in util.ancestry(old_path)):
```

Reply:
I know about that flake8 warning, I did see it when running tox. Though I decided to ignore it as it is a bug. [[1]](https://github.com/PyCQA/pycodestyle/issues/513) [[2]](https://github.com/PyCQA/pycodestyle/issues/498) [[3]](https://github.com/PyCQA/pycodestyle/pull/499) [[4]](https://gitlab.com/pycqa/flake8/issues/139) 

PEP8 was updated to allow both stiles and states breaking before (like I did) even as the preferred stile for new code. [[5]](https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator)

For those reasons I suggest we ignore this bogus warning by adding the following line to the flake8 configuration in `tox.ini`:
```
ignore = W503
````

Reply:
Got it; thanks for digging! I agree: let’s ignore this warning. Would you mind adding that ignore flag to this PR?